Sources_v3/StreamChat/Database/DTOs/MessageReactionDTO.swift

```diff
@@ -40,6 +40,32 @@ extension MessageReactionDTO {
         return try? context.fetch(request).first
     }
     
+    static func loadCurrentUserReactions(
+        for messageId: MessageId,
+        context: NSManagedObjectContext
+    ) -> [MessageReactionDTO] {
+        let request = NSFetchRequest<MessageReactionDTO>(entityName: MessageReactionDTO.entityName)
+        request.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [
+            NSPredicate(format: "message.id == %@", messageId),
+            NSPredicate(format: "user.currentUser != nil")
```

Comment:
Where does the `user.currentUser` come from? I couldn't find it anywhere in the PR 🤔

Reply:
`user.currentUser` is the reverse relation for `currentUser.user`. We don't expose this relation via property but it exists in the db model.

Reply:
You can't use a nasty trick like this and not even mention it in a comment! 😅 I find this very dangerous, to be honest. I think a more _boring_ approach with getting the current user DTO first and use it's userId would be better here. @b-onc ?

Reply:
I wasn't aware this could be done 🤯 Agreed though, let's keep it in the back of our heads (that this is also a viable solution) but my vote is for the boring approach.

Reply:
fixed in bf46e6ec6d59ac8a40deb6a06142e0ff179fe5e2